### PR TITLE
Data Source: azurerm_virtual_network_gateway - support new property private_ip_address

### PR DESCRIPTION
**Pull request by neil-yechenwei**
fixes https://github.com/hashicorp/terraform-provider-azurerm/issues/21424

image

### CODE REVIEW

File: internal/services/network/virtual_network_gateway_data_source.go [hunk 1]
```diff
@@ -326,6 +331,10 @@ func flattenVirtualNetworkGatewayDataSourceIPConfigurations(ipConfigs *[]network
 				}
 			}
 
+			if privateIP := props.PrivateIPAddress; privateIP != nil {
+				v["private_ip_address"] = *privateIP
+			}
```
Comment (katbyte) [inline internal/services/network/virtual_network_gateway_data_source.go:336]: do we want to always set this? ie
```suggestion
			
				v["private_ip_address"] = pointer.From(props.PrivateIPAddress)
			
```

Comment (neil-yechenwei) [reply internal/services/network/virtual_network_gateway_data_source.go:336]: Only when private_ip_address_enabled is true, private_ip_address has value. So we may need nil check.

Comment (neil-yechenwei) [reply internal/services/network/virtual_network_gateway_data_source.go]: Updated

**Comment by neil-yechenwei** [comment]
@katbyte ,thanks for your comment. I updated PR. Please take another look. Thanks.

**Review by stephybun — APPROVED** [review]
Thanks @neil-yechenwei LGTM 🍰

**Pull request merged**